src/StoppingCondition.jl

```diff
@@ -0,0 +1,86 @@
+export AddStoppingCondition
+
+
+function AddStoppingCondition(p::DSProblem, c::T) where T <: AbstractStoppingCondition
+    push!(p.stoppingconditions, c)
+end
+
+CheckStoppingConditions(p::DSProblem) = CheckStoppingConditions(p, p.stoppingconditions)
+function CheckStoppingConditions(p::DSProblem, c::Vector{T}) where T <: AbstractStoppingCondition
+    for condition in c
+        #Stopping conditions return true if optimisation should continue,
+        #false if the condition is met and optimisation should stop
+        if CheckStoppingCondition(p, condition) == false
+            SetStatus(p, condition)
+            return false
+        end
+    end
+
+    return true
+end
+
+function SetStatus(p, s::T) where T <: AbstractStoppingCondition
+    p.status.optimization_status = StoppingConditionStatus(s)
+end
+
+StoppingConditionStatus(::T) where T <: AbstractStoppingCondition = "Unknown stopping condition status"
+
+function SetupStoppingConditions(p::DSProblem)
+    for c in p.stoppingconditions
+        SetupStoppingCondition(p, c)
+    end
+end
+
+SetupStoppingCondition(p::DSProblem, ::AbstractStoppingCondition) = nothing
+
+#===== Built-in stopping conditions =====#
+
+#Iteration limit
+mutable struct IterationStoppingCondition <: AbstractStoppingCondition
+    limit::Int64
+end
+
+StoppingConditionStatus(::IterationStoppingCondition) = "Iteration limit"
+
+CheckStoppingCondition(p::DSProblem, s::IterationStoppingCondition) = p.status.iteration < s.limit
+
+function SetupStoppingCondition(p::DSProblem, s::IterationStoppingCondition)
```

Comment:
I think I prefer the term `init` for these rather than setup, because it should already be setup by the time the optimization is called and we are just initializing the relevant fields for execution here.

Reply:
Agreed, renamed in b3ef240.